partcad/src/partcad/part_factory_stl.py

```
@@ -33,8 +35,64 @@
+                from OCP.BRep import BRep_Builder
+                from OCP.TopoDS import TopoDS_Face, TopoDS_Shell, TopoDS_Solid
+                from OCP.TopAbs import TopAbs_FACE
+                from OCP.BRepCheck import BRepCheck_Shell, BRepCheck_Solid, BRepCheck_Status
+                from OCP.BRepGProp import BRepGProp
+                from OCP.GProp import GProp_GProps
+
+                builder = BRep_Builder()
+                reader = RWStl.ReadFile_s(os.fsdecode(self.path))
+
+                if not reader:
+                    raise RuntimeError("Failed to read the STL file")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)